src/app.py

```diff
@@ -158,14 +158,14 @@ def __call__(self, user: User = Depends(current_user)):
             default_roles = [
                 role.strip() for role in auth_default_roles.split(",")
             ]
-        all_user_roles = list(set(user_roles + default_roles))
+        all_roles = list(set(user_roles + default_roles))
+        all_roles = sum((expand_role(role) for role in all_roles), [])
```

Comment:
Just making sure that the `sum` is here used on purpose? The usage looks super weird to me

Reply:
yes, this is basically concat (summing all the given lists). The idea is to get, well, list of all applicable permissions, instead a list of expanded permissions.

For example:

```
>>> test = [[1, 2, 3], [4, 5, 6], [7, 8]]
>>> sum(test, [])
[1, 2, 3, 4, 5, 6, 7, 8]
```

I'm not aware of a better idiom for concat in python. If that's unclear I can rewrite it to explicit for loop, or a bit complex list comprehension - I think this will work in this case too:

```
[perm for role in all_roles for perm in expand(role)]
```